### PR TITLE
Test gRPC client/server

### PR DESCRIPTION
**Pull request by gaschler**
Adds an integration test and a test with a mock MapBuilderInterface to cover client/server communication.

### CODE REVIEW

File: cartographer_grpc/client_server_test.cc [hunk 1]
```diff
@@ -19,11 +19,44 @@
...
+using testing::_;
+
 namespace cartographer_grpc {
 namespace {
 
+constexpr char kSensorId[] = "sensor";
+
+class MockMapBuilder : public cartographer::mapping::MapBuilderInterface {
+ public:
+  MockMapBuilder() = default;
```
Comment (cschuet) [inline cartographer_grpc/client_server_test.cc]: Can't you delete this?

File: cartographer_grpc/client_server_test.cc [hunk 2]
```diff
@@ -42,22 +75,88 @@ class ClientServerTest : public ::testing::Test {
...
 TEST_F(ClientServerTest, StartAndStopServer) {
+  InitializeRealServer();
+  server_->Start();
+  server_->Shutdown();
+}
+
+TEST_F(ClientServerTest, StartAndStop) {
+  InitializeRealServer();
+  server_->Start();
+  InitializeStub();
```
Comment (cschuet) [inline cartographer_grpc/client_server_test.cc]: Did you check initializing the sub actually fails if you don't start the server? Just to make sure this test actually tests something other than what StartAndStopServer above already tests.

File: cartographer_grpc/client_server_test.cc (hunk 1 above)
Comment (gaschler) [reply cartographer_grpc/client_server_test.cc]: Yeah, thanks.

File: cartographer_grpc/client_server_test.cc (hunk 2 above)
Comment (gaschler) [reply cartographer_grpc/client_server_test.cc]: Right, initializing the stub does not require the serve to run, so I remove this test.

**Comment by gaschler** [comment]
@cschuet PTAL at the corrections

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**